src/NUnitTestAdapter/NUnit3TestDiscoverer.cs

```diff
@@ -123,14 +124,26 @@ public void DiscoverTests(IEnumerable<string> sources, IDiscoveryContext discove
                             (new[] {"contains no tests", "Has no TestFixtures"}).Any(msgNode.GetAttribute("value")
                                 .Contains))
                         {
-                            if (Settings.Verbosity>0)
+                            if (Settings.Verbosity > 0)
                                 TestLog.Info("Assembly contains no NUnit 3.0 tests: " + sourceAssembly);
                         }
                         else
                             TestLog.Info("NUnit failed to load " + sourceAssembly);
                     }
 
                 }
+                catch (NUnitEngineException e)
+                {
+                    if (e.InnerException is BadImageFormatException)
+                    {
+                        // we skip the native c++ binaries that we don't support.
+                        TestLog.Warning("Assembly not supported: " + sourceAssembly);
+                    }
+                    else
+                    {
+                        TestLog.Warning("Exception thrown discovering tests in " + sourceAssembly, e);
+                    }
+                }
```

Comment:
In the latest engine, an NUnitEngineException is thrown instead of a BadImageFormatException for native assemblies. That's seems like 'more correct' behaviour on the engine's part, so I've updated the adapter to match. Glad there was a unit test to catch this! 😄